tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,201 @@
+import logging
+import pytest
+
+from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
+from ocs_ci.utility.deployment_openshift_logging import (
+    check_health_of_clusterlogging, get_clusterlogging_subscription
+)
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocp_upgrade
+)
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.utils import (
+    TimeoutSampler, get_ocp_version, run_cmd
+)
+
+logger = logging.getLogger(__name__)
+
+
+def check_cluster_logging():
+    """
+    Few checks to assert for before and after upgrade
+    1. Checks for pods in openshift-logging project
+    2. Checks for the health of logging cluster before upgrade
+
+    """
+
+    assert get_all_pods(
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    ), 'Some or all pods missing in namespace'
+
+    assert check_health_of_clusterlogging(), "Cluster is not Healthy"
+
+
+def check_csv_logging_upgrade_phase():
+    """
+    The function checks in the CSV for the phase "Succeeded"
+    for cluster-logging operator and Elastic-search
+    operator
+
+    Returns:
+        bool: True if CSV status is Succeeded, False otherwise
+
+    """
+
+    csv = ocp.OCP(
+        kind=constants.CLUSTER_SERVICE_VERSION,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    csv_after_upgrade = csv.get(out_yaml_format=True)
+    clo_phase_after_upgrade = (
+        csv_after_upgrade.get('items')[0].get('status').get('phase')
```

Comment:
I am concerned here with indexes of items as it can take some time till there are 2 CSVs when upgrade start rolling.

Also the second one CSV will disappear once the upgrade finish AFAIK.

Also not sure if we can rely that 1st item will be always the first CSV as I don't know how OCS returns those. 

As in the packageManifest you should be able to get current (default) CSV name for specific channel you should access CSV directly with it's name IMO.